tests/js/session-tests.js

```diff
@@ -929,5 +929,31 @@ module.exports = {
             });
             realm.close();
         });
+    },
+
+    async testAnalyticsSubmission() {
+        const context = node_require('realm/package.json');
+        const analytics = node_require('realm/lib/submit-analytics');
+
+        const payload = await analytics.fetchPlatformData(context, 'TestEvent')
+        .catch((e) => {
+            return Promise.reject("Failed to fetch platform data  [Error: " + e + "]");
+        });
+
+        TestCase.assertDefined(payload.webHook);
+        TestCase.assertType(payload.webHook.event, 'string');
+        TestCase.assertDefined(payload.webHook.properties);
+        TestCase.assertType(payload.webHook.properties.Binding, 'string');
+        TestCase.assertDefined(payload.mixPanel);
+        TestCase.assertType(payload.mixPanel.event, 'string');
+        TestCase.assertDefined(payload.mixPanel.properties);
+        TestCase.assertType(payload.mixPanel.properties.Binding, 'string');
+
+        await analytics.submitStageAnalytics('TestEvent')
+        .catch((e) => {
+            return Promise.reject("Failed to submit webhook analytics  [" + e + "]")
+        });
```

Comment:
Again, since we're not expecting this to happen in the test and this would only benefit our developer in debugging a future issue, I would probably just leave this `.catch` out.
```suggestion
        await analytics.submitStageAnalytics('TestEvent');
```